CMakeLists.txt

```diff
@@ -51,6 +56,53 @@ if (ENDIAN)
   add_compile_definitions(ARCH_IS_BIG_ENDIAN=1)
 endif()
 
+if(USE_FOXGLOVE_SDK)
+  if (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    set(FOXGLOVE_SDK_RELEASES "https://github.com/foxglove/foxglove-sdk/releases/download/sdk%2Fv0.9.0/foxglove-v0.9.0-cpp-aarch64-unknown-linux-gnu.zip" "a554f82b3d1c3096457887bccee7088646b729494ec18909846ac7fcdd06ee1d")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+    set(FOXGLOVE_SDK_RELEASES "https://github.com/foxglove/foxglove-sdk/releases/download/sdk%2Fv0.9.0/foxglove-v0.9.0-cpp-x86_64-unknown-linux-gnu.zip" "48ea17c348583154d7c1934c9f7e82f57067e2afa76b586e1f7ac14cc6b6c1d0")
+  else()
+    message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
+  endif()
+
+  list(GET FOXGLOVE_SDK_RELEASES 0 url)
+  list(GET FOXGLOVE_SDK_RELEASES 1 hash)
+
+  # Download Foxglove SDK shared lib
+  include(FetchContent)
+  # Download the SDK
+  FetchContent_Declare(
+    foxglove_sdk
+    URL ${url}
+    URL_HASH SHA256=${hash}
+  )
+  FetchContent_MakeAvailable(foxglove_sdk)
+
+  file(GLOB_RECURSE _foxglove_sdk_sources ${foxglove_sdk_SOURCE_DIR}/src/*.cpp)
+  add_library(foxglove_sdk SHARED
+    ${_foxglove_sdk_sources}
+  )
+  file(GLOB_RECURSE _foxglove_sdk_headers ${foxglove_sdk_SOURCE_DIR}/include/*.hpp)
+  target_include_directories(foxglove_sdk
+    PUBLIC
+      $<BUILD_INTERFACE:${foxglove_sdk_SOURCE_DIR}/include>
+      $<INSTALL_INTERFACE:include>
+  )
```

Comment:
wonder if we should try to make this match closer with https://docs.foxglove.dev/docs/sdk -- or if there were reasons that didn't work?  (although I like your GLOB_RECURSE better than the GLOB in docs...maybe we should update the docs?)